src/services/firebase.ts

```diff
@@ -112,7 +118,54 @@ export namespace ourFirebase {
     // Make sure we listen to match changes only once.
     // 'gamePortal/matches'
     // store.dispatch(updateMatchList);
-    db().ref('gamePortal/matches' + matchIds); // TODO
+    let tempMatchIds: Promise<MatchInfo>[] = [];
+    for (let matchId of matchIds) {
+      tempMatchIds.push(getMatchDetail(matchId));
+    }
+    Promise.all(tempMatchIds).then((matches: MatchInfo[]) => {
```

Comment:
Always filter the failed promise. There is not guarantee that one of `matches` is not error.